extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisorIOConfigTest.java

```diff
@@ -171,7 +171,7 @@ public void testSerdeForConsumerPropertiesWithPasswords() throws Exception
     String jsonStr = "{\n"
                      + "  \"type\": \"kafka\",\n"
                      + "  \"topic\": \"my-topic\",\n"
-                     + "  \"consumerProperties\": {\"bootstrap.servers\":\"localhost:9092\",\n"
+                     + "  \"consumerProperties\": {\"bootstrap.servers\":{\"type\": \"default\", \"password\": \"localhost:9092\"},\n"
```

Comment:
This part is going to throw a wrench into backwards compatibility (we'll be writing ioConfigs that prior versions can't read). It might negatively affect rolling upgrades.

I think the easiest way to solve this is to adjust DefaultPasswordProvider so it serializes as a simple string. (It already is able to deserialize itself from a simple string, so no problem on that end.) The way to do that is add a `@JsonValue` annotation to the `getPassword()` method.

Reply:
Ah that's a good point (in order to get the DefaultPasswordProvider to serialize back to a simple string I think we also need to add `@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)` so it won't serialize the type info). The tricky piece here is not breaking the `PasswordProviderRedactionMixIn`, as it seems that `@JsonValue` takes precedence over `@JsonIgnore`, even when I set `@JsonValue(false)` in the redaction mixin. I'll whip up a custom serializer to handle this.

Reply:
Excellent point about the redaction mixin. Looking forward to what you come up with.